contrib/node/tests/python/pants_test/contrib/node/tasks/test_node_resolve_src_deps.py

```diff
@@ -0,0 +1,361 @@
+ 
+    dep_node_path = node_paths.node_path(dep)
+    app_node_path = node_paths.node_path(app)
+    self.assertIsNotNone(dep_node_path)
+    self.assertIsNotNone(app_node_path)
+ 
+    # Verify that 'app/node_modules' has a correct symlink to 'dep'
+    app_node_modules_path = os.path.join(app_node_path, 'node_modules')
+    link_dep_path = os.path.join(app_node_modules_path, 'dep')
+    self.assertTrue(os.path.exists(app_node_modules_path))
+    self.assertTrue(os.path.islink(link_dep_path))
+
+    expected = os.path.relpath(dep_node_path, app_node_modules_path)
+    self.assertEqual(os.readlink(link_dep_path), expected)
+ 
+    # Verify that 'app/node_modules/.bin' has a correct symlink to 'dep'
+    app_bin_dep_path = os.path.join(dep_node_path, 'cli.js')
+    dep_bin_path = os.path.join(app_node_path, 'node_modules', '.bin')
+    link_dep_bin_path = os.path.join(dep_bin_path, 'dep')
+
+    self.assertTrue(os.path.islink(link_dep_bin_path))
+    relative_path = os.readlink(link_dep_bin_path)
+    expected = os.path.relpath(app_bin_dep_path, dep_bin_path)
+    self.assertEqual(relative_path, expected)
+ 
+  def test_resolve_symlink_bin_dict(self):
+    dep = self._create_dep(provide_bin=True, use_bin_dict=True)
+    app = self._create_app(dep)
+    _, node_paths = self._resolve_target(app)
+ 
+    dep_node_path = node_paths.node_path(dep)
+    app_node_path = node_paths.node_path(app)
+    self.assertIsNotNone(dep_node_path)
+    self.assertIsNotNone(app_node_path)
+ 
+    # Verify that 'app/node_modules/.bin' has a correct symlink to 'dep_cli'
+    app_bin_dep_path = os.path.join(dep_node_path, 'cli.js')
+    dep_bin_path = os.path.join(app_node_path, 'node_modules', '.bin')
+    link_dep_bin_path = os.path.join(dep_bin_path, 'dep_cli')
+    self.assertTrue(os.path.islink(link_dep_bin_path))
+    relative_path = os.readlink(link_dep_bin_path)
+    expected = os.path.relpath(app_bin_dep_path, dep_bin_path)
+    self.assertEqual(relative_path, expected)
+ 
+  def test_resolve_transitive_deps(self):
+    trans_dep = self._create_trans_dep()
+    dep = self._create_dep(trans_dep=trans_dep, provide_bin=True, use_bin_dict=True)
+    app = self._create_app(dep)
+    task, node_paths = self._resolve_target(app)
+ 
+    trans_dep_node_path = node_paths.node_path(trans_dep)
+    dep_node_path = node_paths.node_path(dep)
+    app_node_path = node_paths.node_path(app)
+    self.assertIsNotNone(trans_dep_node_path)
+    self.assertIsNotNone(dep_node_path)
+    self.assertIsNotNone(app_node_path)
+ 
+    script_path = os.path.join(app_node_path, 'index.js')
+    out = task.node_distribution.node_command(args=[script_path]).check_output()
+    lines = {line.strip() for line in out.splitlines()}
+    self.assertIn('2', lines)
+ 
+  def test_file_dependency_not_found_in_build_graph(self):
+    dep = self._create_dep()
+    app = self._create_app(dep, dep_not_found=True)
+    with self.assertRaises(TaskError):
+      self._resolve_target(app)
```

Comment:
Thanks for the test covering a missing dep! Could you add an assertion about the error message?